src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -18,11 +18,12 @@ public static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
                 return;
 
 #if TARGET_AMD64 || TARGET_ARM64 || TARGET_LOONGARCH64
-            // The exact matrix on when ZeroMemory is faster than InitBlockUnaligned is very complex. The factors to consider include
+            // The exact matrix on when ZeroMemory is faster than SpanHelpers.Fill is very complex. The factors to consider include
             // type of hardware and memory alignment. This threshold was chosen as a good balance across different configurations.
             if (byteLength > 768)
                 goto PInvoke;
-            Unsafe.InitBlockUnaligned(ref b, 0, (uint)byteLength);
+
+            SpanHelpers.Fill(ref b, byteLength, (byte)0);
```

Comment:
Should we rather fix "TODO: Optimize other platforms to be on par with AMD64 CoreCLR" below instead of calling SpanHelpers.Fill?

If we know that we are clearing the memory with all zeros, we should be able to do a bit better than SpanHelpers.Fill.

Reply:
Can it be investigated & improved separately? So we can also see dotnet/performance impact from this change alone (likely, regressions), and then from tunning if needed. As this change will require a lot of measurements on different platforms I presume..

Reply:
Sure, it can be improved separately. I expect that the current version of the change is going to regress perf of number of microbenchmarks so you would see quite a bit of fallout. 

You can copy & paste SpanHelpers.Fill here and delete everything not relevant to zero-fill from it. It should reduce the perf gap.

Reply:
Pushed the copy&paste, although, feels like it could be improved. E.g. it only uses `Vector<T>` which means a gap for 16-32 bytes data. Also, I presume it could use DC ZVA on arm64.